x-pack/plugins/observability_solution/infra/public/pages/metrics/inventory_view/components/waffle/node.tsx

```diff
@@ -60,55 +58,47 @@ export const Node = ({
 
   const showContainerAssetDetailPage = nodeType === 'container' && isContainerAssetViewEnabled;
 
-  const toggleAssetPopover = () => {
+  const toggleAssetPopover = useCallback(() => {
     if (nodeType === 'host' || showContainerAssetDetailPage) {
       setFlyoutUrlState({ detailsItemId: node.id, assetType: nodeType });
     } else {
       togglePopover();
     }
-  };
-
-  const nodeSquare = (
-    <NodeSquare
-      squareSize={squareSize}
-      togglePopover={toggleAssetPopover}
-      showToolTip={showToolTip}
-      hideToolTip={hideToolTip}
-      color={color}
-      nodeName={node.name}
-      value={value}
-      showBorder={detailsItemId === node.id || isPopoverOpen}
-    />
-  );
+  }, [node.id, nodeType, setFlyoutUrlState, showContainerAssetDetailPage, togglePopover]);
 
   return (
     <>
-      {isPopoverOpen ? (
-        <EuiPopover
-          button={nodeSquare}
```

Comment:
Removing this caused an issue with the position of the popover arrow:
After the change: 

https://github.com/user-attachments/assets/216b0609-3c21-4cfd-ba67-2a4c29d44b49


Before:

https://github.com/user-attachments/assets/0fa4cefa-392a-497f-a4ae-ae2730cd0ee7

Reply:
**fixed**, video was attached in the description